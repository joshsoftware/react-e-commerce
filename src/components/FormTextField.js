import React from 'react';
import FormField from './FormField';

const firstname = {
    field: 'exampleFirstName',
    labelText: 'First Name',
    type: 'firstname',
    name: 'firstname',
    placeholder: 'John'
  },
  lastname = {
    field: 'exampleLastName',
    labelText: 'Last Name',
    type: 'lastname',
    name: 'lastname',
    placeholder: 'Doe'
  },
  country = {
    field: 'exampleCountry',
    labelText: 'Country',
    type: 'country',
    name: 'country',
    placeholder: ''
  },
  state = {
    field: 'exampleState',
    labelText: 'State',
    type: 'state',
    name: 'state',
    placeholder: ''
  },
  city = {
    field: 'exampleCity',
    labelText: 'City',
    type: 'city',
    name: 'city',
    placeholder: ''
  },
  address = {
    field: 'exampleAddress',
    labelText: 'Address',
    type: 'address',
    name: 'address',
    placeholder: ''
  };

const FormTextField = () => {
  return (
    <>
      <FormField formfield={firstname} />
      <FormField formfield={lastname} />
      <FormField formfield={country} />
      <FormField formfield={state} />
      <FormField formfield={city} />
      <FormField formfield={address} />
    </>
  );
};

export default FormTextField;
