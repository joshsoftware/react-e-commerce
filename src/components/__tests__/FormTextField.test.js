import { render, cleanup } from '@testing-library/react';
import React from 'react';
import FormTextField from '../FormTextField';

afterEach(cleanup);

describe('FormTextField Component', () => {
  it('Must return FormTextField Component', () => {
    const { asFragment } = render(<FormTextField />);
    expect(asFragment()).toMatchSnapshot();
  });
});
