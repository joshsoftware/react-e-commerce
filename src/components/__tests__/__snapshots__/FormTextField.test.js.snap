// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextField Component Must return FormTextField Component 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleFirstName"
    >
      First Name
    </label>
    <input
      class="form-control"
      name="firstname"
      placeholder="John"
      type="firstname"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleLastName"
    >
      Last Name
    </label>
    <input
      class="form-control"
      name="lastname"
      placeholder="Doe"
      type="lastname"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleCountry"
    >
      Country
    </label>
    <input
      class="form-control"
      name="country"
      placeholder=""
      type="country"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleState"
    >
      State
    </label>
    <input
      class="form-control"
      name="state"
      placeholder=""
      type="state"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleCity"
    >
      City
    </label>
    <input
      class="form-control"
      name="city"
      placeholder=""
      type="city"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleAddress"
    >
      Address
    </label>
    <input
      class="form-control"
      name="address"
      placeholder=""
      type="address"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</DocumentFragment>
`;
